### PR TITLE
Revise core metadata spec

### PR DESCRIPTION
**Pull request by ptrgags**
This branch has my changes to the Cesium 3D Metadata Spec.

Changes so far:

- Addressed many small edits based on feedback
- Did a pass to update the glossary (though there's still some terms I think we need to define better)
- Started removing unnecessary forward references
- Started removing unnecessary sentences for conciseness
- Wrote text for a non-normative overview of how metadata is used in 3D Tiles at various levels of granularity
- Added an appendix to compare this metadata spec to the old batch tables concept 
- pointed out places where diagrams will go, as we need many more

@sanjeetsuhag @ErixenCruz

### CODE REVIEW

File: specification/Metadata/0.0.0/README.md [hunk 1]
```diff
@@ -1209,17 +1267,37 @@ Furthermore, depending on the implementation, the method of specifying a texture
...
+* **component** (of an element) - For elements of type `ARRAY`, the values contained within are called components. In other words, a property array contains elements which contain components.
 * **element** (of a property array) - a single entry of a property array. This stores the value of one property for a single instance.
-* **component** (of an element) - For elements of type `ARRAY`, the values contained within are called components. In other words, a property array contains elements which contain components. 
-* **metadata texture** - A texture that represents a property directly, no instance table needed
-* **encoding** - A method of storing metadata on disk in a format such as JSON, binary, or texture encodings.
\ No newline at end of file
+* **encoding** - A concrete method of storing metadata in a format such as JSON, binary, or textures.
+* **instance** - A concrete encoding of a class consisting of a value for every property of the class.
+* **instance table** - A table of metadata values indexed by instance IDs. The columns of the table are a set of parallel property arrays.
+* **metadata** - Additional application-specific properties in a structured format.
```
Comment (ptrgags) [inline specification/Metadata/0.0.0/README.md:1278]: how exactly do we want to define "metadata"? I don't particularly like the "application-specific" wording, as 

I think of metadata as annotating data with additional data... but not sure how to word that.

File: specification/Metadata/0.0.0/README.md [hunk 2]
```diff
@@ -11,61 +13,123 @@
...
+* whether the wall is in good condition or damaged at that point.
+* Surface temperature of the wall at that point
+
+DIAGRAM: metadata textures
+
+This metadata can be used by the rendering engine for fine-tuned styling. It could also be used in a UI so when a user clicks a point, metadata about that point could be displayed on the screen.
+
+3D Tiles Metadata is designed to support these various levels of granularity
+
+## Overview
```
Comment (ptrgags) [inline specification/Metadata/0.0.0/README.md:129]: Remove this section when finished writing new intro

File: specification/Metadata/0.0.0/README.md [hunk 3]
```diff
@@ -74,50 +138,75 @@ Using terminology from the geospatial industry, **features** are geometric entit
...
+* Boolean values (`BOOLEAN`)
+* Signed integers (`INT8`, `INT16`, `INT32`, `INT64`)
+* Unsigned integers (`INT8`, `INT16`, `INT32`, `INT64`)
+* Floating point numbers (`FLOAT16`, `FLOAT32`, `FLOAT64`)
+* Strings (`STRING`)
 
-The following example shows the basics of how classes describe the data types, without describing where the data is stored. The following section will show how to connect these classes to the actual metadata in various storage formats.
+Even though a string is a sequence of characters, it is considered to be a single indivisible value when storing metadata.
 
+**Array types** (`ARRAY`) are sequences of a single primitive type. The values are stored contiguously. Arrays can be fixed length or variable length.
```
Comment (ptrgags) [inline specification/Metadata/0.0.0/README.md:162]: To be more consistent, do we want to use "variable length" or "variable size"?

In the rest of the document, it seems like "variable-size array" is far more common. Maybe we should go with that?

File: specification/Metadata/0.0.0/README.md (hunk 1 above)
Comment (ptrgags) [inline specification/Metadata/0.0.0/README.md:1278]: Another question about consistent terminology: is metadata "scoped to", "assigned to", "attached to", "associated with", or "added to" data?

Comment (ptrgags) [inline specification/Metadata/0.0.0/README.md:1278]: Also is there a general term for the objects that can have associated metadata? I.e. is there a general term we can use to say "tileset or layer or tile or content or vertex or texel"? It certainly would help when writing.

File: specification/Metadata/0.0.0/README.md [hunk 4]
```diff
@@ -74,50 +138,75 @@ Using terminology from the geospatial industry, **features** are geometric entit
...
+
+For a complete list of vocabulary used, see the [Glossary](#glossary)
 
 ## Concepts
 
-### Classes
+### Data Types
 
-A **class** describes a collection of related pieces of metadata called **properties** and their respective data types. In some ways, this is similar to describing the columns of a database table. However, this analogy is not completely applicable, as this specification allows for texture storage, not just columnar formats.
+**Primitive types** are data types that cannot be divided further. These include:
```
Comment (ptrgags) [inline specification/Metadata/0.0.0/README.md:152]: Are we okay with calling the smallest metadata types (numeric and strings) "primitive types"? I also considered "intrinsic", but I think "primitive" better conveys "you can't divide this further"

(Yes I did consider "atomic" as well, but that has a different meaning in concurrent programming)

File: specification/Metadata/0.0.0/README.md [hunk 5]
```diff
@@ -177,20 +268,33 @@ Below is an example of a well-formed instance table representing a class.
...
+
+The metadata values are stored in parallel arrays called **property arrays**, one per column. Each property array stores values for a single property. the `i-th` value of each property array is the value of that property for the instance with an instance ID of `i`.
+
+DIAGRAM: Property arrays
+
+### Encodings
+
+#### Binary Encoding
+
+TODO: This section depends on buffers/buffer views
```
Comment (ptrgags) [inline specification/Metadata/0.0.0/README.md:287]: Can this specification reference the binary buffers spec for this?

**Review by ptrgags — COMMENTED** [review]
Left some notes about outstanding questions I've encountered when writing the edits so far.

**Comment by lilleyse** [comment]
These revisions have been incorporated into https://github.com/CesiumGS/3d-tiles/blob/3d-tiles-next/specification/Metadata/1.0.0/README.md

**Pull request closed without merging**